src/asphalt/core/_event.py

```diff
@@ -59,29 +70,75 @@ def __repr__(self) -> str:
         )
 
 
-T_Event = TypeVar("T_Event", bound=Event)
+@dataclass
+class Signal(Generic[T_Event]):
+    """
+    Declaration of a signal that can be used to dispatch events.
 
+    This is a descriptor that returns itself on class level attribute access and a bound
+    version of itself on instance level access. Streaming events and dispatching
```

Comment:
Actually, maybe this is better:
```suggestion
    version of itself on instance level access. Dispatching and streaming
```